MapleServer2/PacketHandlers/Game/PartyHandler.cs

```diff
@@ -147,6 +179,24 @@ private void HandleJoin(GameSession session, PacketReader packet)
             }
         }
 
+        private void HandleLeave(GameSession session, PacketReader packet)
+        {
+            Party party = GameServer.PartyManager.GetPartyById(session.Player.PartyId);
+            session.Send(PartyPacket.Leave(session.Player, 1)); //1 = You're the player leaving
+            session.Player.PartyId = 0;
+            if (party == null)
+            {
+                return;
+            }
+            party.RemoveMember(session.Player);
+            party.BroadcastPacketParty(PartyPacket.Leave(session.Player, 0));
+            if (party.Leader.CharacterId == session.Player.CharacterId)
+            {
+                party.FindNewLeader();
+            }
+            party.CheckDisband();
```

Comment:
`FindNewLeader` and `CheckDisband` both feel like methods that should always happen whenever `party.RemoveMember` is called. If so, this should live inside `Party.cs` instead